### ISSUE TITLE
Crash on SwrveWakefulService when receiving a null intent

### ISSUE THREAD

**Issue by nimeacuerdo** [Swrve/swrve-android-sdk#293]
Hi,

we are seeing crashes in our crash monitoring service for an app that uses version 6.3.0 of the swrve-firebase SDK.

The stacktrace is as follows:

```
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.content.Intent.getIntExtra(java.lang.String, int)' on a null object reference
        at androidx.legacy.content.WakefulBroadcastReceiver.completeWakefulIntent(WakefulBroadcastReceiver:126)
        at com.swrve.sdk.SwrveWakefulService.onHandleIntent(SwrveWakefulService:20)
        at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:67)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:203)
        at android.os.HandlerThread.run(HandlerThread.java:61)
```

Reviewing the source code this crash is somewhat expected, since docs for handleIntent method state that the received intent could be null:

```
    /**
     * This method is invoked on the worker thread with a request to process.
     * Only one Intent is processed at a time, but the processing happens on a
     * worker thread that runs independently from other application logic.
     * So, if this code takes a long time, it will hold up other requests to
     * the same IntentService, but it will not hold up anything else.
     * When all requests have been handled, the IntentService stops itself,
     * so you should not call {@link #stopSelf}.
     *
     * @param intent The value passed to {@link
     *               android.content.Context#startService(Intent)}.
     *               This may be null if the service is being restarted after
     *               its process has gone away; see
     *               {@link android.app.Service#onStartCommand}
     *               for details.
     */
    @WorkerThread
    protected abstract void onHandleIntent(@Nullable Intent intent);
```

`SwrveWakefulService` does not perform any check on this and is blindly capturing the whole Exception:

```
    @Override
    protected void onHandleIntent(Intent intent) {
        try {
            SwrveBackgroundEventSender sender = getBackgroundEventSender();
            sender.handleSendEvents(intent.getExtras());
        } catch (Exception ex) {
            SwrveLogger.e("SwrveWakefulService exception (intent: %s): ", ex, intent);
        } finally {
            SwrveWakefulReceiver.completeWakefulIntent(intent);
        }
    }
```

**Comment by davidbreenwex** [comment]
Hi @nimeacuerdo,

Thank you for reporting the issue. 
We are looking into this now. 
I'll update you as once the fix has been released. 

Best, 
David

**Comment by davidbreenwex** [comment]
Hi @nimeacuerdo,

We have just released a fix for this in version 7.2.1 of the SDK:
https://docs.swrve.com/release-notes/sdk-release-notes/android-sdk-release-notes/#Release_721

Thanks again for reporting the issue and providing the error logs. 

Best, 
David
